dao/src/main/java/com/garret/movies/dao/util/MovieUtil.java

```diff
@@ -0,0 +1,117 @@
+package com.garret.movies.dao.util;
+
+
+import com.garret.movies.dao.entity.Actor;
+import com.garret.movies.dao.entity.Country;
+import com.garret.movies.dao.entity.Genre;
+import com.garret.movies.dao.entity.Language;
+import com.garret.movies.dao.exception.IncorrectDateException;
+import lombok.extern.slf4j.Slf4j;
+
+import java.sql.Date;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+@Slf4j
+public class MovieUtil {
+
+    private static final SimpleDateFormat formatter = new SimpleDateFormat("dd MMM yyyy", Locale.ENGLISH);
+    private static final Pattern DATE_PATTERN =
+            Pattern.compile("^(([0-9])|([0-2][0-9])|([3][0-1])) (Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec) \\d{4}$");
```

Comment:
@Garret-vin Looks odd, what this for?

Reply:
@mmarusich this pattern checks for correct input date for parsing in method parseDate()